test/runtests.jl

```diff
@@ -1,4 +1,5 @@
 using Test, LinearAlgebra, HssMatrices
+BLAS.set_num_threads(2)
```

Comment:
what does this do here?

Reply:
Using too many threads when combining BLAS operations with Julia multithreading results in slowdowns due to CPU oversubscription. I put 2 here as a compromise, 1 could make more sense.

Reply:
Wouldn't this affect "normal" operation where multithreading is off?

So if I run the old and new code both without multithreading, would I see a slowdown?

Reply:
Same benchmark, but using multithreaded AppleAccelerate as BLAS, its threads play more nicely with Julia, the results are even better: 
![speedup_plot](https://github.com/user-attachments/assets/50314c71-913f-4247-a9c4-bc4946b47b31)